### ISSUE TITLE
Update BagIt Version to 1.0

### ISSUE THREAD

**Issue by horsburgh** [hydroshare/hydroshare#4031]
Up to this point, HydroShare has been using Version 0.9.6 of the BagIt specification for encoding bags for download. There is a Version 1.0 of BagIt that was released in 2018. We should consider updating. 

https://tools.ietf.org/html/rfc8493 

After reviewing the newer specification, it appears that the differences are minor. The only real difference I could find was in which checksum algorithm should be used. Here is the relevant text from Section 2.4 of the document:

> Starting with BagIt 1.0, bag creation and validation tools MUST support the SHA-256 and SHA-512 algorithms [RFC6234] and SHOULD enable SHA-512 by default when creating new bags.  For backwards compatibility, implementers SHOULD support MD5 [RFC1321] and SHA-1 [RFC3174].  Implementers are encouraged to simplify the process of adding additional manifests using new algorithms to streamline the process of in-place upgrades.

I suggest leaving all currently generated bags as they are, although we may want to consider whether we want to regenerate all bags using BagIt 1.0 the next time they are downloaded. When a new bag is created, it should use BagIt 1.0 and should use the SHA-512 checksum algorithm.

So, it looks like the following would be needed to update the new bag generation process:
1. Update the checksum algorithm to use SHA-512 and ensure that the manifest file is named "manifest-sha-512.txt"
2. Update the "bagit.txt" file to include the correct version number
3. Update the name of the main manifest file (manifest-sha-512.txt) in the tag manifest file (tagmanifest-sha-512.txt). The new tag manifest should be called "tagmanifest-sha-512.txt" and should list the "manifest-sha-512.txt", "bagit.txt", and "readme.txt" files.
4. Update the URL to the BagIt specification in the "readme.txt" file to https://tools.ietf.org/html/rfc8493
5. Update the names of "tagmanifest-md5.txt" and "manifest-md5.txt" near the bottom of the "readme.txt" file
6. A developer should scan the BagIt spec and make sure we are not violating any of the file naming rules.

Otherwise everything else stays the same.

**Comment by horsburgh** [comment]
@hyi - You probably want to look at and comment on this as the change in checksum algorithm may have some implication on how we are doing things related to iRODs.

**Comment by hyi** [comment]
@horsburgh Yeah, the bagit files are currently created using an iRODS rule written using iRODS rule language. We will have to update this iRODS rule to update bagit files to conform to version 1.0. Is SHA-512 a requirement for bagit v1.0? It looks like iRODS only supports the base64 encoded sha256 checksum as well as md5, so if SHA-512 is a requirement for bagit v1.0, i.e., SHA256 checksum does not meet the requirement, this may be a showstopper at this point. I can double check with iRODS folks, but I took a quick look at iRODS documentation, and it does not look like iRODS support SHA512 checksums.

**Comment by horsburgh** [comment]
@hyi - I found the language from the spec a little confusing. It says:

> Starting with BagIt 1.0, bag creation and validation tools **MUST** support the SHA-256 and SHA-512 algorithms and **SHOULD** enable SHA-512 by default when creating new bags."

It's not clear why an implementation **MUST** support both SHA-256 and SHA-512 if you can just choose one and use it consistently. If you interpret this strictly, we wouldn't be able to comply because we could not use the SHA-512 algorithm, even though iRODs is capable of using the SHA-256 algorithm.

If iRODs can't do SHA-512, we can get by the **SHOULD** part by using SHA-256. As long as we are always dealing with Bags created by HydroShare using SHA-256 we would not have any problems. If we try to ingest a Bag written by another system that uses the SHA-512 algorithm, we would be in trouble (which is probably why the spec wants tools to support both).
